path_config.go

```diff
@@ -188,9 +204,49 @@ func (b *azureSecretBackend) pathConfigWrite(ctx context.Context, req *logical.R
 		return logical.ErrorResponse(merr.Error()), nil
 	}
 
+	// set up rotation after everything is fine
+	var rotOp string
+	if config.ShouldDeregisterRotationJob() {
+		rotOp = "deregistration"
+		// Ensure de-registering only occurs on updates and if
+		// a credential has actually been registered (rotation_period or rotation_schedule is set)
+		deregisterReq := &rotation.RotationJobDeregisterRequest{
+			MountPoint: req.MountPoint,
+			ReqPath:    req.Path,
+		}
+
+		err := b.System().DeregisterRotationJob(ctx, deregisterReq)
+		if err != nil {
+			return logical.ErrorResponse("error de-registering rotation job: %s", err), nil
+		}
+
+	} else if config.ShouldRegisterRotationJob() {
+		rotOp = "registeration"
```

Comment:
should we use the constants?

Reply:
We can use the constants but it requires us to move off the sdk tag. The values between plugins don't have to be the same for any technical reason although I agree this should become the constants at some point.